docs/git/index.md

```diff
@@ -0,0 +1,295 @@
+
+Try to find all of the differences between these three documents. Can you work out which one is the finished one which was meant to go on the website?
+
+You probably found it hard to see all of the differences (some were really small, like adding or removing a comma!), and non-obvious which is the most complete version!
+
+### Exercise 2 (10 minutes)
+
+Get into groups of three. One of your group should make a shopping list of 5 items, and send the list to the group.
+
+The other two people in the group should each add a few items to the list, and also remove one item, and then send back the whole new list to the person who made the list.
+
+The first person should merge the new lists, so that there is one shopping list. The items people removed should not be in the final list.
+
+Report back to the class what your experience was like. What was easy? What was frustrating?
+
+## How does Git help?
+
+Let's start by taking the shopping list problem, and applying it to our classes.
+
+Imagine your teacher wanted everyone in the class to answer three questions and to send back the answers. What capabilities would we need in order to be able to do that?
+
+**First**, the teacher needs to be able to write the questions, and **store** them somewhere.
+
+When we're using Git, we write things down in files in a folder (these could be text files, Word documents, images, or really any kind of file). When we save these files, we say we're _committing_ them, and we call the folder where we're saving them a _repository_ (or _repo_ for short).
+
+TODO: Insert graphic of teacher writing and committing.
+
+**Next**, the teacher needs to put the repository somewhere where the learners can get it. We call this _pushing_ the repository. The teacher won't send a copy to each learner, but will put one copy somewhere on the Internet, and tell the learners where it is. The place the teachers on this course will be pushing to is a website called [GitHub](https://github.com), but there are other websites they could push to if they wanted to. This is the difference between Git and GitHub - Git is a way of storing and sharing files, and GitHub is one website where you can use Git.
+
+**Then**, now that the teacher has pushed the questions, each learner needs to be able to **get the questions onto their computer**.
+
+When we're using Git, we call this _cloning_ the teacher's repository (because we're making our own copy). After we've cloned the repository, we will have the same folder on our computer as the teacher created, committed, and pushed.
+
+TODO: Insert sequence diagram of teacher committing, pushing, and learners cloning.
+
+## The Git Cheatsheet
+
+We tend to do the same four or five things in Git over and over again, but it can be easy to forget them. There's a handy reference at (TODO: Link to cheat sheet) to help you remember. We'll use it in our next exercise:
+
+### Exercise 3 (15 minutes)
```

Comment:
There's a nice phrase that I've heard about how to teach: "I do, we do, you do". The idea being that we're going from the teacher, to the student with supervision, to the student by themselves.

I therefore wonder if it would be useful to demo these steps first? However, it's possible we could get away with skipping this if we're short on time. The TA model somewhat mixes up the first 2 stages.

Reply:
I was thinking that the video from the cheatsheet was kind of an "I do", but it probably _is_ a pretty big step. Maybe we could encourage the teacher to open up the cheatsheet and play the video, narrating a bit as it goes?

Reply:
Yeah, that's fair 👍 Kind of what I was getting at with the TA comment, it's possible it would come out naturally.

Reply:
Added a prompt to show the cheatsheet